File: challenge_app/lib/domain/usecase/task/create_task_usecase.dart
import 'package:challenge_app/domain/models/task/gateway/task_gateway.dart';
import 'package:challenge_app/domain/models/task/task.dart';

class CreateTaskUseCase {
  final TaskGateway _taskGateway;
  CreateTaskUseCase(this._taskGateway);
  
  Future<Task> createNewTask(Task newTask) async => _taskGateway.createTask(newTask);
}

File: challengeWeb/src/app/domain/usecase/task/create_task_usecase.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TaskGateway } from '../../models/task/gateway/task_gateway';
import { Task } from '../../models/task/task';

@Injectable({providedIn: 'root'})
export class CreateTaskUseCase {
    private _taskGateway: TaskGateway;

    constructor(taskGateWay: TaskGateway) {
        this._taskGateway = taskGateWay;
    }

    createNewTask(task: Task): Observable<Task> {
        return this._taskGateway.createTask(task);
    }
}